homeassistant/components/history.py

```diff
@@ -233,7 +232,12 @@ def get(self, request, datetime=None):
         if start_time > now:
             return self.json([])
 
-        end_time = start_time + one_day
+        end_time = request.GET.get('end_time')
+        if end_time:
+            end_time = dt_util.as_utc(
+                dt_util.parse_datetime(end_time))
```

Comment:
Let's add the same check here as https://github.com/home-assistant/home-assistant/pull/5895/files#diff-bfb762edec66809f6ea72fe4dd388975R221